src/ngp_algorithms/SSTTAMSAveragesAlg.C

```diff
@@ -0,0 +1,312 @@
+    turbKineticEnergy_(get_field_ordinal(realm.meta_data(), "turbulent_ke")),
+    specDissipationRate_(
+      get_field_ordinal(realm.meta_data(), "specific_dissipation_rate")),
+    avgVelocity_(get_field_ordinal(realm.meta_data(), "average_velocity")),
+    avgDudx_(get_field_ordinal(realm.meta_data(), "average_dudx")),
+    avgTkeRes_(get_field_ordinal(realm.meta_data(), "average_tke_resolved")),
+    avgProd_(get_field_ordinal(realm.meta_data(), "average_production")),
+    avgTime_(get_field_ordinal(realm.meta_data(), "rans_time_scale")),
+    avgResAdeq_(
+      get_field_ordinal(realm.meta_data(), "avg_res_adequacy_parameter")),
+    tvisc_(get_field_ordinal(realm.meta_data(), "turbulent_viscosity")),
+    alpha_(get_field_ordinal(realm.meta_data(), "k_ratio")),
+    Mij_(get_field_ordinal(realm.meta_data(), "metric_tensor"))
+{
+}
+
+void
+SSTTAMSAveragesAlg::execute()
+{
+  using Traits = nalu_ngp::NGPMeshTraits<ngp::Mesh>;
+
+  const auto& meta = realm_.meta_data();
+  const DblType dt = realm_.get_time_step();
+  const int nDim = meta.spatial_dimension();
+
+  const stk::mesh::Selector sel =
+    (meta.locally_owned_part() | meta.globally_shared_part()) &
+    stk::mesh::selectField(
+      *meta.get_field(stk::topology::NODE_RANK, "average_velocity"));
+
+  const auto& meshInfo = realm_.mesh_info();
+  const auto ngpMesh = meshInfo.ngp_mesh();
+  const auto& fieldMgr = meshInfo.ngp_field_manager();
+
+  const auto tvisc = fieldMgr.get_field<double>(tvisc_);
+  const auto tke = fieldMgr.get_field<double>(turbKineticEnergy_);
+  const auto sdr = fieldMgr.get_field<double>(specDissipationRate_);
+  const auto density = fieldMgr.get_field<double>(density_);
+  auto alpha = fieldMgr.get_field<double>(alpha_);
+  auto avgProd = fieldMgr.get_field<double>(avgProd_);
+  auto avgTkeRes = fieldMgr.get_field<double>(avgTkeRes_);
+  auto avgTime = fieldMgr.get_field<double>(avgTime_);
+  auto resAdeq = fieldMgr.get_field<double>(resAdeq_);
+  auto avgResAdeq = fieldMgr.get_field<double>(avgResAdeq_);
+  const auto vel = fieldMgr.get_field<double>(velocity_);
+  const auto dudx = fieldMgr.get_field<double>(dudx_);
+  auto avgVel = fieldMgr.get_field<double>(avgVelocity_);
+  auto avgDudx = fieldMgr.get_field<double>(avgDudx_);
+  const auto Mij = fieldMgr.get_field<double>(Mij_);
+
+  const DblType betaStar = betaStar_;
+  const DblType CMdeg = CMdeg_;
+
+  nalu_ngp::run_entity_algorithm(
+    "SSTTAMSAveragesAlg_computeAverages",
+    ngpMesh, stk::topology::NODE_RANK, sel,
+    KOKKOS_LAMBDA(const Traits::MeshIndex& mi) {
+      // Calculate alpha
+      if (tke.get(mi, 0) == 0.0)
+        alpha.get(mi, 0) = 1.0;
+      else {
+        alpha.get(mi, 0) = 1.0 - avgTkeRes.get(mi, 0) / tke.get(mi, 0);
+
+        // limiters
+        alpha.get(mi, 0) = std::min(alpha.get(mi, 0), 1.0);
+
+        // FIXME: What to do with a_kol in SST?
+        const double a_kol = 0.01;
```

Comment:
you can make this static constexpr in the class just to make it obvious that this is a numerical tweak that might need some thought.

Reply:
moved to class